src/main/java/org/apache/commons/cli/help/HelpFormatter.java

```diff
@@ -0,0 +1,286 @@
+/*
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+package org.apache.commons.cli.help;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.function.Function;
+
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Util;
+
+/**
+ * A default formatter implementation for standard usage.
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * Options options = new Options();
+ * options.addOption(OptionBuilder.withLongOpt("file").withDescription("The file to be processed").hasArg().withArgName("FILE").isRequired().create('f'));
+ * options.addOption(OptionBuilder.withLongOpt("version").withDescription("Print the version of the application").create('v'));
+ * options.addOption(OptionBuilder.withLongOpt("help").create('h'));
+ *
+ * String header = "Do something useful with an input file";
+ * String footer = "Please report issues at https://example.com/issues";
+ *
+ * HelpFormatter formatter = new HelpFormatter();
+ * formatter.printHelp("myapp", header, options, footer, true);
+ * </pre>
+ * <p>
+ * This produces the following output:
+ * </p>
+ * <pre>
+ * usage: myapp -f &lt;FILE&gt; [-h] [-v]
+ * Do something useful with an input file
+ *
+ *  -f,--file &lt;FILE&gt;   The file to be processed
+ *  -h,--help
+ *  -v,--version       Print the version of the application
+ *
+ * Please report issues at https://example.com/issues
+ * </pre>
+ */
+public class HelpFormatter extends AbstractHelpFormatter {
```

Comment:
We would have 2 classes called `HelpFormatter`. I think we need to deprecate the old one, it's too confusing otherwise.

Reply:
Check.